packages/xo-server/src/api/xostor.mjs

```diff
@@ -0,0 +1,184 @@
+  Thin: 'thin',
+  Thick: 'thick',
+}
+const LV_NAME = 'thin_device'
+const PROVISIONING = Object.values(ENUM_PROVISIONING)
+const VG_NAME = 'linstor_group'
+const _XOSTOR_DEPENDENCIES = ['xcp-ng-release-linstor', 'xcp-ng-linstor']
+const XOSTOR_DEPENDENCIES = _XOSTOR_DEPENDENCIES.join(',')
+
+const log = createLogger('xo:api:pool')
+
+function pluginCall(xapi, host, plugin, fnName, args) {
+  return xapi.call('host.call_plugin', host._xapiRef, plugin, fnName, args)
+}
+
+async function destroyVolumeGroup(xapi, host, force) {
+  const isVgAlreadyExist =
+    Object.keys(
+      JSON.parse(
+        await pluginCall(xapi, host, 'lvm.py', 'list_volume_groups', {
+          vg_name: VG_NAME,
+        })
+      )
+    ).length > 0
+  if (isVgAlreadyExist) {
+    log.info(`Trying to delete the ${VG_NAME} volume group.`, { hostId: host.id })
+    return pluginCall(xapi, host, 'lvm.py', 'destroy_volume_group', {
+      vg_name: VG_NAME,
+      force: String(force),
+    })
+  }
+}
+
+async function installOrUpdateDependencies(host, method = 'install') {
+  if (method !== 'install' && method !== 'update') {
+    throw new Error('Invalid method')
+  }
+
+  const xapi = this.getXapi(host)
+  log.info(`Trying to ${method} XOSTOR dependencies (${XOSTOR_DEPENDENCIES})`, { hostId: host.id })
+  for (const _package of _XOSTOR_DEPENDENCIES) {
+    await xapi.call('host.call_plugin', host._xapiRef, 'updater.py', method, {
+      packages: _package,
+    })
+  }
+}
+
+export function installDependencies({ host }) {
+  return installOrUpdateDependencies.call(this, host)
+}
+installDependencies.description = 'Install XOSTOR dependencies'
+installDependencies.permission = 'admin'
+installDependencies.params = {
+  hostId: { type: 'string' },
+}
+installDependencies.resolve = {
+  host: ['hostId', 'host', 'administrate'],
+}
+
+export function updateDependencies({ host }) {
+  return installOrUpdateDependencies.call(this, host, 'update')
+}
+updateDependencies.description = 'Update XOSTOR dependencies'
+updateDependencies.permission = 'admin'
+updateDependencies.params = {
+  hostId: { type: 'string' },
+}
+updateDependencies.resolve = {
+  host: ['hostId', 'host', 'administrate'],
+}
+
+export async function formatDisks({ disks, force, host, ignoreFileSystems, provisioning }) {
+  const rawDisks = disks.join(',')
+  const xapi = this.getXapi(host)
+
+  const lvmPlugin = (fnName, args) => pluginCall(xapi, host, 'lvm.py', fnName, args)
+  log.info(`Format disks (${rawDisks}) with force: ${force}`, { hostId: host.id })
+
+  if (force) {
+    await destroyVolumeGroup(xapi, host, force)
+  }
+  await lvmPlugin('create_physical_volume', {
+    devices: rawDisks,
+    ignore_existing_filesystems: String(ignoreFileSystems),
+    force: String(force),
+  })
+
+  await lvmPlugin('create_volume_group', {
+    devices: rawDisks,
+    vg_name: VG_NAME,
+  })
+
+  if (provisioning === ENUM_PROVISIONING.Thin) {
+    await lvmPlugin('create_thin_pool', {
```

Comment:
At this point in the code, that doesn't seem possible to me.

The `virtual group` "linstor_group" and the `logical volumes` attached to this group are deleted when `xostor.destroy` is called. However, if the user decides to delete the storage from another source, at the XOSTOR creation the `force` option will need to be set to `true` in order to delete the volume group and the attached logical volumes forcibly, otherwise, the `formatDisks` function will throw an error during the `create_physical_volume` call `([...] Can't initialize physical volume "/dev/sdb" of volume group "linstor_group" without -ff\n...)`.
BTW, ATM, there is a limitation to 1 XOSTOR/pool.